### ISSUE TITLE
wxGrid -- wxGridCellDateTimeRenderer() constructor wrong

### ISSUE THREAD

**Issue by jdgelti** [wxWidgets/wxWidgets#22279]
Hi developers,
I'm currently using wxWidgets 3.1.5 (on Windows).
I think there is a bug in include/wx/generic/gridctrl.h, line 220 -- at least for UNICODE compilations:
```
    wxGridCellDateTimeRenderer(const wxString& outformat = wxASCII_STR(wxDefaultDateTimeFormat),
                               const wxString& informat = wxASCII_STR(wxDefaultDateTimeFormat));
```
Since `wxDefaultDateTimeFormat` is defined as `wxT("%c")` the above constructor resolves to
```
wxGridCellDateTimeRenderer(const wxString& outformat = wxString::FromAscii(L"%c"),
                           const wxString& informat = wxString::FromAscii(L"%c") );
```
but `wxString::FromAscii()` does not have an overload which accepts a `wchar_t[]`

I think, the constructor could be
```
    wxGridCellDateTimeRenderer(const wxString& outformat = wxString(wxDefaultDateTimeFormat),
                               const wxString& informat = wxString(wxDefaultDateTimeFormat));
```
to fix the issue.

**Comment by vadz** [comment]
Sorry, what is exactly the problem here, i.e. what is the error message you're getting?

Also, why do you say that

> `wxDefaultDateTimeFormat` is defined as `wxT("%c")`

when this is not the case, it's declared as

```cpp
extern WXDLLIMPEXP_DATA_BASE(const char) wxDefaultDateTimeFormat[];
```

Are you using unmodified wxWidgets sources?

**Comment by jdgelti** [comment]
Visual Studio 2015 compilation (of my application, which uses wxGrid):
```
...\include\wx/generic/gridctrl.h(220): error C2665: 'wxString::FromAscii': none of the 5 overloads could convert all the argument types
...\include\wx/string.h(1658): note: could be 'wxString wxString::FromAscii(const unsigned char *)'
...\include\wx/string.h(1646): note: or       'wxString wxString::FromAscii(char)'
...\include\wx/string.h(1645): note: or       'wxString wxString::FromAscii(const char *)'
...\include\wx/generic/gridctrl.h(220): note: while trying to match the argument list '(const wchar_t [3])'
```
There is a compile error, because there is no `wxString::FromAscii()` method which takes a `const wchar_t[]`; this is what `wxT("%c")` emits when compiling in Unicode.

**However**, as I see now: The error does not come from wxWidgets code itself, but from 'wxSheet' (a library which we have used back then). There is `wx24desf.h` which contains (line 66):
```
#ifndef wxSL_INVERSE // !wxCHECK_VERSION(2,5,3) // FIXME temp fix until 2.5.4
    #define wxDefaultDateTimeFormat wxT("%c")
#endif // wxInvalidDateTime
```
and **THAT** is the cause of the error... -- Sorry for bothering you.

**Comment by vadz** [comment]
Yes, this fragment redefining `wxDefaultDateTimeFormat` just needs to be removed.
